### ISSUE TITLE
CSOM ApplyProvisioningTemplate with parameters

### ISSUE THREAD

**Issue by jmmartinez84** [SharePoint/PnP-Sites-Core#1887]
Hi there, 
I'm able to pass parameters to provisioning template using powershell script using -Parameters option
`Apply-PnPProvisioningTemplate -Web $web -Path $path -Parameters @{"xxx"=$xxx;"yyy"=$yyy;"zzz"=$zz;}`
But I don't kno how to do the same using CSOM
`newWeb.ApplyProvisioningTemplate(template, info);`

thanks in advance

**Comment by siata13** [comment]
Hi,
I think that you can use 'Parameters' property directly in ProvisioningTemplate object:
![image](https://user-images.githubusercontent.com/6162213/53234315-91738100-368f-11e9-93ca-5657e308995c.png)

**Comment by SpliceVW** [comment]
> I think that you can use 'Parameters' property directly in ProvisioningTemplate object:

The Parameters property is read-only on the `ProvisioningTemplate` object.

@jmmartinez84 did you ever find a solution? I am in a similar situation where I need to add parameters at runtime.

**Comment by siata13** [comment]
@SpliceVW Yes, it is read-only (getter), but you can still use it to add your parameters:
`pnpTemplate.Parameters.Add(key,value);`
